packages/firmware_notifier/lib/firmware_notifier.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:collection/collection.dart';
+import 'package:desktop_notifications/desktop_notifications.dart';
+import 'package:fwupd/fwupd.dart';
+
+Future<Map<FwupdDevice, FwupdRelease>> getUpdates() async {
+  final fwupdClient = FwupdClient();
+  await fwupdClient.connect();
+
+  final devices = await fwupdClient.getDevices();
+  final updates = {
+    for (final device in devices)
+      device: (await fwupdClient.getReleases(device.deviceId).catchError(
+              (_) => <FwupdRelease>[],
+              test: (e) =>
+                  e is FwupdNothingToDoException ||
+                  e is FwupdNotSupportedException))
+          .firstWhereOrNull(
+        (r) => r.flags.contains(FwupdReleaseFlag.isUpgrade),
+      )
+  }..removeWhere((_, update) => update == null);
+  await fwupdClient.close();
+  return Map<FwupdDevice, FwupdRelease>.from(updates);
+}
+
+Future<List<Notification>> sendUpdateNotifications(
+    NotificationsClient client) async {
+  final updates = await getUpdates();
+
+  return Future.wait(updates.entries.map(
+    (e) => client.notify(
+      'Firmware update available for ${e.key.name}',
```

Comment:
o-ou, how do we localize this :thinking:

Reply:
Hehe 😌 
I wonder how snapd does this on its own? 
When you install a snap via the preview software we send a notification with the response from snapd change (install, remove, refresh, connect) which are magically localised

Reply:
Hm, flutter_localizations probably doesn't work in a pure dart package. I've tried adding [intl_translation](https://pub.dev/packages/intl_translation), but there's a version conflict with the dbus package :/
```
The current Dart SDK version is 2.18.2.

Because desktop_notifications >=0.6.3 depends on dbus ^0.7.0 which depends on args ^2.0.0, desktop_notifications >=0.6.3 requires args ^2.0.0.
And because intl_translation >=0.17.0 depends on args >=0.12.1 <2.0.0 and intl_translation <0.17.0 requires SDK version >=1.12.0 <2.0.0, desktop_notifications >=0.6.3 is incompatible with
  intl_translation.
So, because firmware_notifier depends on both desktop_notifications ^0.6.3 and intl_translation any, version solving failed.
```

Any ideas?

Reply:
we can try our luck overriding the args version in firmware_notifier/pubspec.yaml - might work if they're not using any of the breaking changes?

```yaml
dependency_overrides:
  args: ^2.0.0
```

Reply:
there's home hope that the conflicts will be sorted out soon :)

https://github.com/dart-lang/intl_translation/issues/167

Reply:
looks like args is not enough but there's a conflicting transitive petitparser dependency too:
```yaml
dependency_overrides:
  args: ^2.0.0
  petitparser: ^5.0.0
```
or alternatively, use the latest int_translation straight from git until it gets released:
```yaml
intl_translation:
  git:
    url: https://github.com/dart-lang/intl_translation.git
    ref: a543412
```

Reply:
btw we can leave this to a separate step unless you prefer to figure it out before taking the first step :)

Reply:
Yes, totally - I'm still struggling to get this to work. The first solution isn't great since the current release of intl_translation is quite old. With the git version I'm not getting any sensible output from my .arb file so far.. :/